compiler/rustc_next_trait_solver/src/solve/assembly/mod.rs

```diff
@@ -522,8 +523,8 @@ where
         // Instead of adding the logic here, it's a better idea to add it in
         // `EvalCtxt::disqualify_auto_trait_candidate_due_to_possible_impl` in
         // `solve::trait_goals` instead.
-        let result = if let Err(guar) = goal.predicate.error_reported() {
-            G::consider_error_guaranteed_candidate(self, guar)
+        let result = if let ty::Error(guar) = goal.predicate.self_ty().kind() {
+            G::consider_error_guaranteed_candidate(self, goal, guar)
```

Comment:
I think we should move this out of `assemble_builtin_impl_candidate` to avoid ambiguity between e.g. blanket impls and the builtin impl candidate :thinking: